drivers/soundwire/cadence_master.c

```diff
@@ -368,6 +372,97 @@ static int cdns_hw_reset(void *data, u64 value)
 
 DEFINE_DEBUGFS_ATTRIBUTE(cdns_hw_reset_fops, NULL, cdns_hw_reset, "%llu\n");
 
+static int cdns_parity_error_injection(void *data, u64 value)
+{
+	struct sdw_cdns *cdns = data;
+	struct sdw_slave *slave;
+	struct sdw_bus *bus;
+	int ret;
+
+	if (value != 1)
+		return -EINVAL;
+
+	bus = &cdns->bus;
+
+	/*
+	 * wake up master and slave(s)
+	 */
+	list_for_each_entry(slave, &bus->slaves, node) {
+		/*
+		 * discard devices that are defined in ACPI tables but
+		 * not physically present
+		 */
+		if (slave->dev_num_sticky) {
+			ret = pm_runtime_get_sync(&slave->dev);
+			if (ret < 0 && ret != -EACCES) {
+				dev_err_ratelimited(cdns->dev,
+						    "pm_runtime_get_sync failed in %s, ret %d\n",
+						    __func__, ret);
+				pm_runtime_put_noidle(&slave->dev);
```

Comment:
and this is one of the effects of that crudeness: if one of resumes fails, all previously successfully resumed devices will have unbalanced rtpm and will stay enabled now?

Reply:
yep, that was bad. will remove this.